indexing-service/src/main/java/org/apache/druid/indexing/common/task/IndexTaskUtils.java

```diff
@@ -129,6 +130,7 @@ public static void setTaskDimensions(final ServiceMetricEvent.Builder metricBuil
         DruidMetrics.TAGS,
         task.<Map<String, Object>>getContextValue(DruidMetrics.TAGS)
     );
+    metricBuilder.setDimensionIfNotNull(DruidMetrics.GROUP_ID, task.getGroupId());
   }
 
   public static void setTaskStatusDimensions(
```

Comment:
I think you will want to set groupId in here as well

Reply:
it looks to me like this is called to set the status only when the task completes, and the other ones are called when the task starts, do we need to set group id twice?

Reply:
I suspect that there are different metricBuilder objects being passed to these 2 functions. It should only need to be set once on an object.

Reply:
This method is currently being used in just one place, while emitting the metric `task/run/time`.
You can add the `groupId` dimension to this one as well, just in case you want to group this metric on `groupId`. 

To set the groupId though, you would need to pass the `Task` object to this method, because `TaskStatus` does not seem to have a `groupId` field.

Reply:
hmm, i haven't updated anything but when i run locally task/run/time has groupId appended, am I missing something?

`"feed":"metrics","taskType":"kill","metric":"task/run/time","service":"druid/coordinator","groupId":"api-issued_kill_wikipedia_gklcinen_1000-01-01T00:00:00.000Z_2023-06-13T00:00:00.000Z_2023-06-14T15:34:21.080Z","host":"localhost:8081","version":"27.0.0-SNAPSHOT","value":32189,"dataSource":"wikipedia","taskId":"api-issued_kill_wikipedia_gklcinen_1000-01-01T00:00:00.000Z_2023-06-13T00:00:00.000Z_2023-06-14T15:34:21.080Z","taskStatus":"SUCCESS","timestamp":"2023-06-14T15:34:53.920Z"}`

Reply:
it still seems like in the case of task/run/time the metricsBuilder object will get groupId set with the IndexTaskUtils.setTaskDimensions call in attachCallbacks

Reply:
@suneet-s i added the logic in ForkingTaskRunner, but not in the other two b/c those seem to only apply to streaming tasks which I don't think we really want groupId on, lmk if that's okay with you

Reply:
i had to add the dimension in a couple new places after testing locally, but I tested every scenario where I updated the docs so this should be good to go now, looking into any unit tests i need to change now